crypto/crypto.go

```diff
@@ -122,12 +124,13 @@ func FromECDSA(priv *ecdsa.PrivateKey) []byte {
 	return math.PaddedBigBytes(priv.D, priv.Params().BitSize/8)
 }
 
-func ToECDSAPub(pub []byte) *ecdsa.PublicKey {
-	if len(pub) == 0 {
```

Comment:
There seems to be no unit test that covers the case when `len(pub) == 0`. This is a problem, because even though a 0-length `pub` will return `nil` in the current implementation, we don't know how things could change if the key and signature schemes change because they turn out to be unsafe.